src/client/dfs/dfs.c

```diff
@@ -4262,16 +4367,138 @@ dfs_chmod(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode)
 	return rc;
 }
 
+int
+dfs_chown(dfs_t *dfs, dfs_obj_t *parent, const char *name, uid_t uid, gid_t gid, int flags)
```

Comment:
Is there a performance advantage over having this as a call to dfs_lookup(), dfs_osetattr(), dfs_release()?  If not then we should probably do that to avoid having two code-paths to test?

Reply:
setattr does an extra stat call

Reply:
you implemented osetattr i believe and added a stat there to set the attr and return an updated stat struct. probably a dfuse requirement? adding this flag means an API change, or maybe just another field in the attr flags, which isn't pretty. anyway, yes not in scope here..

Reply:
Returning the stat structure is a requirement of fuse yes.  I was just wondering if we could avoid having two implementations to test/maintain but it seems that we can't today.